source/agora/network/NetworkManager.d

```diff
@@ -893,6 +895,37 @@ public class NetworkManager
         }
     }
 
+    /***************************************************************************
+
+        Retrieve blocks starting from block_height up to the highest block
+        that's available from the connected nodes.
+
+    ***************************************************************************/
+
+    private void getUnknownTXs (Ledger ledger) @safe nothrow
+    {
+        auto unknown_txs = ledger.getUnknownTXHashes();
+
+        foreach (peer; this.peers[])
+        {
+            if (unknown_txs.length == 0)
+                break;
+
+            foreach (tx; peer.getTransactions(unknown_txs))
+            {
+                try
+                {
+                    ledger.acceptTransaction(tx);
```

Comment:
You need to check the return value.

Reply:
Do we want to log it, if its invalid? I could not think of any other reason for checking if it is valid or not.

Reply:
You can't think of any? Check the next line.

If the tx is invalid then the missing tx set should not be modified..

Reply:
Wait I just realized something.

Reply:
My assumption was wrong. In fact if the tx is invalid there is no other tx with a matching hash, that would make it a hash collision.

Reply:
I wonder if this can be abused in some way. If a bad actor nominates a big tx set of invalid hashes, will that cause the node to constantly try to download these txes?

Reply:
by invalid I mean fake hashes of non-existing txes.

Reply:
It was hard to follow but, yes :)

Reply:
> I wonder if this can be abused in some way. If a bad actor nominates a big tx set of invalid hashes, will that cause the node to constantly try to download these txes?
 
 Oops I didnt see the last 2 messages when I replied to you.
 
Until the end of that round, yes. We clear the unknown txs at the end of each round. "big tx set" will probably be bound by some upper limit too eventually. Maybe the block size or tx count?

Reply:
We can may be give up on looking for hashes after a certain amount of tries

Reply:
Anyway it shouldn't block the PR.